src/Morphir/TypeScript/Backend.elm

```diff
@@ -81,93 +83,155 @@ mapModuleDefinition opt distribution currentPackagePath currentModulePath access
             , typeDefs = typeDefs
             }
     in
-    [moduleUnit]
+    [ moduleUnit ]
+
+
+{-| Map a Morphir Constructor (A tuple of Name and Constructor Args) to a Typescript AST Interface
```

Comment:
I suggest rewording this:

> Map one Constructor from a Morphir custom type to a TypeScript interface.

Reply:
That suggests that the input is a Morphir custom type, and the output is a Typescript interface. Strictly speaking, the input is only part of a Morphir custom type.

At the time I also found it really helpful to clarify that a constructor consists of a tuple of (Name, List[args]) because that isn't clearly documented anywhere else in our code base. That sort of structural information was what confused me the most when I first tried to understand the code Attila had written.

I suppose it's not accurate any more though (now there's also the variables to consider). So you might be right that that needs to change.

Reply:
> Strictly speaking, the input is only part of a Morphir custom type.

True.

> At the time I also found it really helpful to clarify that a constructor consists of a tuple of (Name, List[args]) because that isn't clearly documented anywhere else in our code base. That sort of structural information was what confused me the most when I first tried to understand the code Attila had written.

It's the job of the type system and variable names to provide this info. I agree the function declaration is hard to read, but can we fix by adding some meaningful type aliases rather than a comment?

Reply:
That sounds good